sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -2,20 +2,24 @@
 
 import io.sentry.core.protocol.User;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.CopyOnWriteArrayList;
 
 public class Scope implements Cloneable {
   private SentryLevel level;
   private String transaction;
   private User user;
   private List<String> fingerprint = new ArrayList<>();
-  private List<Breadcrumb> breadcrumbs = new CopyOnWriteArrayList<>();
+  private final CircularFifoQueue<Breadcrumb> breadcrumbs;
```

Comment:
I'd use type `Queue` here

Reply:
The base type is `AbstractCollection` though.

Reply:
extends `AbstractCollection` but implements `Queue`, so it should be `Queue`